bundles/org.eclipse.emfcloud.modelserver.emf/src/org/eclipse/emfcloud/modelserver/emf/common/SchemaController.java

```diff
@@ -21,11 +21,30 @@
    @Inject
    private ModelRepository modelRepository;
 
+   @Inject
+   private SchemaRepository schemaRepository;
+
    public void getSchema(final Context ctx, final String modeluri) {
       this.modelRepository.getModel(modeluri).ifPresentOrElse(
          instance -> ctx.json(JsonResponse.success(JsonSchema.from(instance.eClass()))),
          () -> ctx.status(404)
             .json(JsonResponse.error(String.format("Schema for '%s' not found!", ctx.queryParam("modeluri")))));
    }
 
+   public void getTypeSchema(final Context ctx, final String modeluri) {
+      this.modelRepository.getModel(modeluri).ifPresentOrElse(
+         instance -> ctx.json(JsonResponse.success(JsonSchema.from(instance))),
+         () -> ctx.status(404)
+            .json(JsonResponse.error(String.format("Type schema for '%s' not found!", ctx.queryParam("modeluri")))));
+   }
+
+   public void getJsonFormsUISchema(final Context ctx, final String schemaname) {
```

Comment:
Why does this method name have the specific 'JsonForms' in the name? Just wondering.

Reply:
I agree, this prefix can be removed as it does not add any value to the readability of the method's name ;)